### PR TITLE
Armory entry logging

### PR DESCRIPTION
**Pull request by flappybatpal**
[ADMIN][FEATURE]

## About the PR 
Logs when a player enters an unauthorized armory.
120 second cooldown to catch repeat entry without spam.
Does not log an authorized armory on the assumption that entry after this point is legitimate.

## Why's this needed? 
It's not always clear if a player runs into/teleports/uses disposals or otherwise enters the armory.
Picking up equipment in there is not necessarily logged so there may be no evidence that someone actually entered and took things from the armory.
Initially I did not log traitors but decided it's useful to know either way.

### CODE REVIEW

File: code/area.dm [hunk 1]
```diff
@@ -3544,6 +3544,30 @@ ABSTRACT_TYPE(/area/station/ai_monitored/storage/)
...
+	var/static/list/entered_ckeys = list()
+
+	Entered(atom/movable/A, atom/oldloc)
+		. = ..()
+		if (current_state < GAME_STATE_FINISHED)
+			if(istype(A, /mob/living))
+				var/mob/living/M = A
+				if(!M.client)
+					return
+				if(!M.client.holder)
```
Comment (TobleroneSwordfish) [inline code/area.dm:3556]:
```suggestion
				if(M.client.holder)
```
to ignore admins

File: code/area.dm [hunk 2]
```diff
@@ -3544,6 +3544,30 @@ ABSTRACT_TYPE(/area/station/ai_monitored/storage/)
...
+				if (M.mind && M.mind.assigned_role == "Head of Security")
+					return
+				var/armory_auth = FALSE
+				for_by_tcl(O, /obj/machinery/computer/riotgear)
+					if (O.authed)
+						armory_auth = TRUE
+						break
+				entered_ckeys += M.client.ckey
+				SPAWN(120 SECONDS)
+					entered_ckeys -= entered_ckeys[1]
```
Comment (TobleroneSwordfish) [inline code/area.dm:3569]: probably better to remove the specific ckey rather than just the first one, could cause jank?

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

File: code/area.dm (hunk 1 above)
Comment (flappybatpal) [reply code/area.dm:3556]: Leftover from testing!

File: code/area.dm (hunk 2 above)
Comment (flappybatpal) [reply code/area.dm:3569]: I understood the order wouldn't change?

File: code/area.dm [hunk 3]
```diff
@@ -3544,6 +3544,31 @@ ABSTRACT_TYPE(/area/station/ai_monitored/storage/)
 	teleport_blocked = 1
 	spy_secure_area = TRUE
 	station_map_colour = MAPC_ARMOURY
+	var/static/list/entered_ckeys = list()
+
+	Entered(atom/movable/A, atom/oldloc)
+		. = ..()
+		if (current_state < GAME_STATE_FINISHED)
+			if(istype(A, /mob/living))
+				var/mob/living/M = A
```
Comment (Tarmunora) [inline code/area.dm:3553]: check for `living/intangible`

**Review by Tarmunora — CHANGES_REQUESTED** [review]

**Review by Tarmunora — APPROVED** [review]

**Comment by flappybatpal** [comment]
!merge_upstream

**Pull request merged**